example/src/plugins/job-ui-single/pages/CreateJobEntity.tsx

```diff
@@ -54,31 +51,63 @@ export const CreateJobEntity = (props: TCreateJobEntityProps) => {
         outputTarget: defaultJobOutputTarget,
       }
     }
-    if (destinationIsAPackage) {
-      DmssApi.documentAdd({
-        address: jobEntityDestination,
-        document: JSON.stringify(jobEntityFormData),
+
+    const updateDocument = (
+      jobEntityDestination: string,
+      jobEntityFormData: TJob
+    ) => {
+      DmssApi.documentUpdate({
+        idAddress: jobEntityDestination,
+        data: JSON.stringify(jobEntityFormData),
       })
-        .then((response: AxiosResponse) => {
-          onCreate(`${dataSourceId}/$${response.data.uid}`)
+        .then(() => {
+          onCreate(jobEntityDestination)
           setCreatedJobEntity(jobEntityFormData)
         })
         .catch((error: AxiosError<ErrorResponse>) => {
           console.error(error.response?.data)
         })
-    } else {
-      DmssApi.documentUpdate({
-        idAddress: jobEntityDestination,
-        data: JSON.stringify(jobEntityFormData),
+    }
+
+    const addDocument = (
+      jobEntityDestination: string,
+      jobEntityFormData: TJob
+    ) => {
+      DmssApi.documentAdd({
+        address: jobEntityDestination,
+        document: JSON.stringify(jobEntityFormData),
       })
-        .then(() => {
-          onCreate(jobEntityDestination)
+        .then((response: AxiosResponse) => {
+          onCreate(`${dataSourceId}/$${response.data.uid}`)
           setCreatedJobEntity(jobEntityFormData)
         })
         .catch((error: AxiosError<ErrorResponse>) => {
           console.error(error.response?.data)
         })
     }
+
+    const addOrUpdateDocument = (
+      jobExists: boolean,
+      jobEntityDestination,
+      jobEntityFormData
+    ) => {
+      if (jobExists) {
+        updateDocument(jobEntityDestination, jobEntityFormData)
+      } else {
+        addDocument(jobEntityDestination, jobEntityFormData)
+      }
+    }
+
+    // TODO: Implement document check
+    DmssApi.documentCheck({
+      address: jobEntityDestination,
```

Comment:
Can the address point to a list? In that case, we should use addDocument even if the list exist, I think